### ISSUE TITLE
Mocked HTTP request is canceled (sometimes)

### ISSUE THREAD

**Issue by pmarchezz** [DevExpress/testcafe#4550]
### What is your Test Scenario?
Testing a web application doing some XHR requests to fetch data (and mocking them).
 
### What is the Current behavior?
Mocked request inconsistently fails as canceled (in Chrome).
 
### What is the Expected behavior?
Mocked request always returns desired values as configured in test code.
 
### What is your web application and your TestCafe test code?
I managed to reproduce the issue with this simple HTML page with jQuery (real test page does not use jQuery, I used it to reproduce in a minimal example). Example page is hosted on a local Apache webserver on my machine (although original issue was reproduced by several colleagues using different OSes/webservers).

Your website URL (or attach your complete example):
```html
<!DOCTYPE html>
<html>

<head>
    <meta charset="UTF-8">
    <title>TEST</title>
</head>

<body>
    <script src="https://code.jquery.com/jquery-3.4.1.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>

    <button id="clickme">click me</button>

    <script>
        $('button').on("click", function() {
            $.getJSON("http://example.com")
                .done(function(resp) {
                    $("body").append('<div>' + JSON.stringify(resp) + '</div>');
                });
        });
    </script>
</body>
</html>
```
<details>
<summary>Your complete test code (or attach your test files):</summary>

```js
import { Selector, RequestMock } from 'testcafe';

let mock = RequestMock()
	.onRequestTo(new RegExp('example'))
	.respond({"testcafe": "isAwesome"}, 200, {"access-control-allow-origin": "*"});
	
fixture `Getting Started`
	.page `http://localhost/testcafe.html`
	.requestHooks(mock);

test('some test', async t => {
	await t
		.click("button")
		.expect(Selector("body").withText("testcafe").exists).ok();
});
```
</details>

<details>
<summary>Your complete configuration file (if any):</summary>
No configuration
</details>

<details>
<summary>Your complete test report:</summary>

```
 Running tests in:
 - Chrome 78.0.3904.108 / Windows 10

 Getting Started

----
Chrome 78.0.3904.108 / Windows 10
DEBUGGER PAUSE ON FAILED TEST:
AssertionError: expected false to be truthy

Browser: Chrome 78.0.3904.108 / Windows 10

9 |    .requestHooks(mock);
10 |
11 |test('some test', async t => {
12 |    await t
13 |            .click("button")
> 14 |          .expect(Selector("body").withText("testcafe").exists).ok();
15 |});

at <anonymous> (C:\dev\test\test.js:14:57)
at test (C:\dev\test\test.js:11:1)
at <anonymous> (C:\nvm\v10.15.3\node_modules\testcafe\src\api\wrap-test-function
.js:17:28)
at TestRun._executeTestFn (C:\nvm\v10.15.3\node_modules\testcafe\src\test-run\in
dex.js:291:19)
at TestRun.start (C:\nvm\v10.15.3\node_modules\testcafe\src\test-run\index.js:34
2:24)
----

```
</details>

<details>
<summary>Screenshots:</summary>

![testcafe-failure](https://user-images.githubusercontent.com/5880566/70226537-4ecbf280-1751-11ea-8f7f-20a7673cb39a.PNG)

</details>

### Steps to Reproduce:

1. Host the attached HTML file on some random webserver
2. Change the test code to point to correct URL
3. (since the issue is not consistent) repeatedly execute test (I used this command to have some debugging available when it happens: `while true; do testcafe 'chrome --auto-open-devtools-for-tabs' test.js --debug-on-fail; done`)
4. After some successful executions, testcafe fails as in screenshot attached.
 
### Your Environment details:
 
* testcafe version:                   1.7.0
* node.js version:                    10.15.3
* command-line arguments:  testcafe 'chrome --auto-open-devtools-for-tabs' test.js --debug-on-fail
* browser name and version: Chrome 78.0.3904.108
* platform and version:          Windows 10
* other:                                   Tests launched using git bash (pretty sure it doesn't matter)

If any other info is required, please let me know. Thank you in advance.

**Comment by miherlosev** [comment]
Hi @pmarchezz 

I cannot reproduce the problem. Try to add the `fail` handler for the `$.getJSON` method call to inspect possible request problems.
![image](https://user-images.githubusercontent.com/4133518/70232466-91032d00-176d-11ea-933c-dc6f828fa830.png)

**Comment by pmarchezz** [comment]
Hi @miherlosev, thank you for your time and help. I tried adding the fail handler (logging arguments that are passed to the fail function), the result is the following:

```js
$('button').on("click", function() {
	$.getJSON("http://example.com")
		.done(function(resp) {
			$("body").append('<div>' + JSON.stringify(resp) + '</div>');
		})
		.fail(function() {
			console.log(arguments);			
		});
});

```

![testcafe-failure-2](https://user-images.githubusercontent.com/5880566/70233490-f4399300-175e-11ea-9e21-a7b5a367f4a4.PNG)

Let me know if I can help you with debugging in any way. To give you an idea, I reproduced the failure after ~20 successful runs (it may take a while to get there...).
